egs/librispeech/ASR/pruned_transducer_stateless7/multidataset.py

```diff
@@ -25,29 +25,53 @@
 
 
 class MultiDataset:
-    def __init__(self, manifest_dir: str):
+    def __init__(self, manifest_dir: str, cv_manifest_dir: str):
         """
         Args:
           manifest_dir:
             It is expected to contain the following files:
 
-            - multidataset_split_1998/multidataset/multidataset_cuts_train.*.jsonl.gz
+            - librispeech_cuts_train-all-shuf.jsonl.gz
+            - gigaspeech_XL_split_2000/gigaspeech_cuts_XL.*.jsonl.gz
+
+          cv_manifest_dir:
+            It is expected to contain the following files:
+
+            - cv-en_cuts_train.jsonl.gz
         """
         self.manifest_dir = Path(manifest_dir)
+        self.cv_manifest_dir = Path(cv_manifest_dir)
 
     def train_cuts(self) -> CutSet:
         logging.info("About to get multidataset train cuts")
 
+        # LibriSpeech
+        logging.info(f"Loading LibriSpeech in lazy mode")
+        librispeech_cuts = load_manifest_lazy(
+            self.manifest_dir / "librispeech_cuts_train-all-shuf.jsonl.gz"
+        )
+
+        # GigaSpeech
         filenames = glob.glob(
-            f"{self.manifest_dir}/multidataset_split_1998/multidataset/multidataset_cuts_train.*.jsonl.gz"
+            f"{self.manifest_dir}/gigaspeech_XL_split_2000/gigaspeech_cuts_XL.*.jsonl.gz"
         )
 
-        pattern = re.compile(r"multidataset_cuts_train.([0-9]+).jsonl.gz")
+        pattern = re.compile(r"gigaspeech_cuts_XL.([0-9]+).jsonl.gz")
         idx_filenames = ((int(pattern.search(f).group(1)), f) for f in filenames)
         idx_filenames = sorted(idx_filenames, key=lambda x: x[0])
 
         sorted_filenames = [f[1] for f in idx_filenames]
 
-        logging.info(f"Loading {len(sorted_filenames)} splits")
+        logging.info(f"Loading GigaSpeech {len(sorted_filenames)} splits in lazy mode")
+
+        gigaspeech_cuts = lhotse.combine(
+            lhotse.load_manifest_lazy(p) for p in sorted_filenames
+        )
+
+        # CommonVoice
+        logging.info(f"Loading CommonVoice in lazy mode")
+        commonvoice_cuts = load_manifest_lazy(
+            self.cv_manifest_dir / f"cv-en_cuts_train.jsonl.gz"
+        )
 
-        return lhotse.combine(lhotse.load_manifest_lazy(p) for p in sorted_filenames)
+        return CutSet.mux(librispeech_cuts, gigaspeech_cuts, commonvoice_cuts)
```

Comment:
For the best effect you might want to add the following argument to mux: `weights=[len(librispeech_cuts), len(gigaspeech_cuts), len(commonvoice_cuts)]`.

It will lag for a bunch of seconds to scan the number of lines but it'll sample them a bit more uniformly throughout the whole output sequence.

Reply:
Thanks!